### DISCUSSION TITLE
How do I indicate different batches?

### DISCUSSION THREAD

**Discussion by rachelgriffard** [Shenhav-and-Korem-labs/SCRuB#19] [Q&A] [answered]
When using SCRuB, how do I use this function if I have two batches run on different plates? This means there would be multiple "A1" sample well locations within the metadata for samples run across different batches/different plates, for example.

I may be confused on functionality here, and I appreciate any help!

I appreciate any help with this!

**Comment by gaustin15** [comment, marked as answer]
The recommendation for this situation is to run SCRuB separately within each batch, with the idea being that each batch has its own distinct contamination patterns that should be captured in separate SCRuB models. 

So, the code implementing this might look something like:
```r
library(tidyverse)
library(SCRuB)
### assuming a read count matrix `df`, and metadata file `md`
all_scrub_outputs <- list()
for( scrub_batch in unique(md$batch) ){ # looping through all batches
  
    # specify the row names from a particular batch
    batch_indices <- md %>% filter(batch==scrub_batch) %>% row.names()
  
    ## run SCRuB, store resultsin the list
  	all_scrub_outputs[[scrub_batch]] <- SCRuB( df[batch_indices, ], 
  	                                           md[batch_indices, ] %>% 
  	                                                      select(is_control, ## I'm making assumptions about the
  	                                                             sample_type, ## metadata's naming structure 
  	                                                             sample_well) ## which are probably inacurrate in your case
  	                                           )
}
```

Please let me know if I misunderstood the question, or if there’s anything else I can do to help with your implementation.

**Reply by rachelgriffard** [reply]
This is exactly what I meant! Thank you, I appreciate you taking the time to answer this so quickly.
